src/log_surgeon/finite_automata/TaggedTransition.hpp

```diff
@@ -64,11 +61,17 @@ class NegativeTaggedTransition {
         if (state_id_it == state_ids.end()) {
             return std::nullopt;
         }
-        return fmt::format("{}[{}]", state_id_it->second, fmt::join(m_tags, ","));
+
+        if (std::ranges::any_of(m_tags, [](Tag const* tag) { return tag == nullptr; })) {
+            return std::nullopt;
+        }
+        auto const tag_names = m_tags | std::ranges::views::transform(&Tag::get_name);
+
+        return fmt::format("{}[{}]", state_id_it->second, fmt::join(tag_names, ","));
     }
 
 private:
-    std::set<uint32_t> m_tags;
+    std::vector<Tag const*> const m_tags;
```

Comment:
```suggestion
    std::vector<Tag const*> m_tags;
```
Same as the comments above, let's avoid adding const to a member variable

Reply:
Same question as above.